enterprise/internal/codeintel/resolvers/position.go

```diff
@@ -0,0 +1,160 @@
+package resolvers
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"github.com/sourcegraph/go-diff/diff"
+	"github.com/sourcegraph/go-lsp"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/internal/vcs/git"
+)
+
+type positionAdjuster struct {
+	hunks []*diff.Hunk
+}
+
+// newPositionAdjuster creates a positionAdjuster by performing a git diff operation on the give
+// path between the source and target commits. If the commits are the same then a trivial-case
+// no-op position adjuster is returned.
+func newPositionAdjuster(
+	ctx context.Context,
+	repo *types.Repo,
+	sourceCommit string,
+	targetCommit string,
+	path string,
+) (*positionAdjuster, error) {
```

Comment:
```suggestion
func newPositionAdjuster(ctx context.Context, repo *types.Repo, sourceCommit, targetCommit, path string) (*positionAdjuster, error) {
```

As a rule, it's better to let a function signature be long than to split it up like this (at least in Go).
With the types consolidated it's not too bad.
You could also shorten "source" and "target" to "old" and "new" if you wanted to save a bit more horizontal space, YMMV.

Reply:
> As a rule

Why? You would think if it's a rule that gofmt would have something to say about it.

Reply:
Not that kind of rule 🙂. Perhaps I should say: "As a point of style". More in the sense that it's uncommon (at least in Go) and therefore surprising, and thus worth avoiding unless the case is more explicitly unusual than this.